install.sh

```diff
@@ -7,26 +15,78 @@ ORACLE_RPM="$(basename $ORACLE_FILE .zip)"
+  test -f /sbin/chkconfig ||
+    ( echo '#!/bin/sh' | sudo tee /sbin/chkconfig > /dev/null && sudo chmod u+x /sbin/chkconfig )
 
-test -d /var/lock/subsys || sudo mkdir /var/lock/subsys
+  test -d /var/lock/subsys || sudo mkdir /var/lock/subsys
+fi
 
 unzip -j "$(basename $ORACLE_FILE)" "*/$ORACLE_RPM"
-sudo rpm --install --nodeps --nopre "$ORACLE_RPM"
 
-echo 'OS_AUTHENT_PREFIX=""' | sudo tee -a "$ORACLE_HOME/config/scripts/init.ora" > /dev/null
-sudo usermod -aG dba $USER
+if [ $SUDO_ENABLED -eq 1 ]; then
+  sudo rpm --install --nodeps --nopre "$ORACLE_RPM"
+
+  echo 'OS_AUTHENT_PREFIX=""' | sudo tee -a "$ORACLE_HOME/config/scripts/init.ora" > /dev/null
+  sudo usermod -aG dba $USER
+  ( echo ; echo ; echo travis ; echo travis ; echo n ) | sudo AWK='/usr/bin/awk' /etc/init.d/oracle-xe configure
+else
+  mkdir /home/travis/u01
+  rpm --install --nodeps --nopre --noscripts --notriggers --relocate /u01=/home/travis/u01 --relocate /etc=/home/travis/u01 --relocate /usr/=/home/travis/u01 "$ORACLE_RPM"
+
+  # this should check that LD_LIBRARY_PATH was set correctly
+  LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ORACLE_HOME/lib; export LD_LIBRARY_PATH
+  # also ORACLE_SID?
+  ORACLE_SID=XE; export ORACLE_SID
+
+  echo 'OS_AUTHENT_PREFIX=""' | tee -a "$ORACLE_HOME/config/scripts/init.ora" > /dev/null
+  mkdir /home/travis/u01/app/oracle/oradata
+  mkdir /home/travis/u01/app/oracle/diag
+  sed -i "s/%hostname%/localhost/g" $ORACLE_HOME/network/admin/listener.ora
+  sed -i "s/%port%/1521/g" $ORACLE_HOME/network/admin/listener.ora
+  sed -i "s/\/u01/\/home\/travis\/u01/g" $ORACLE_HOME/network/admin/listener.ora
+
+  sed -i "/^memory_target/d" $ORACLE_HOME/config/scripts/init.ora $ORACLE_HOME/config/scripts/initXETemp.ora $ORACLE_HOME/dbs/init.ora
+
+  find $ORACLE_HOME/config -type f | xargs sed -i "s/\/u01/\/home\/travis\/u01/g"
+  find $ORACLE_HOME/config -type f | xargs sed -i "s/%ORACLE_HOME%/\/home\/travis\/u01\/app\/oracle\/product\/11.2.0\/xe/g"
+
+  mkdir -p $ORACLE_HOME/network/log $ORACLE_HOME/config/log
+  touch $ORACLE_HOME/network/log/listener.log $ORACLE_HOME/config/log/CloneRmanRestore.log
+
+  $ORACLE_HOME/bin/lsnrctl start
+  sh -x $ORACLE_HOME/config/scripts/XE.sh
+  cat > $ORACLE_HOME/config/scripts/create_travis_user.sql <<SQL
+host $ORACLE_HOME/bin/orapwd file=$ORACLE_HOME/dbs/orapwXE password=oracle force=y
+connect "SYS"/"oracle" as SYSDBA
+alter user sys identified by "travis";
+alter user system identified by "travis";
+exit
+SQL
+
+  cat $ORACLE_HOME/config/scripts/create_travis_user.sql
+
+  $ORACLE_HOME/bin/sqlplus /nolog @$ORACLE_HOME/config/scripts/create_travis_user.sql
+
+  $ORACLE_HOME/bin/sqlplus sys/travis AS SYSDBA <<SQL
+startup
+SQL
 
-( echo ; echo ; echo travis ; echo travis ; echo n ) | sudo AWK='/usr/bin/awk' /etc/init.d/oracle-xe configure
+fi
 
-"$ORACLE_HOME/bin/sqlplus" -L -S / AS SYSDBA <<SQL
+"$ORACLE_HOME/bin/sqlplus" -L -S sys/travis AS SYSDBA <<SQL
```

Comment:
no, it was 100% necessary.  OS authentication does not appear to be (easily) possible in the container mode, as it requires adding `travis` to the dba group, which require sudo.

However, it may be possible to use `nis` to add `travis` to the dba group in nis, which Oracle may use.  I havent looked into it.